### PR TITLE
BLIP-0011: NameDesc

### PR DESCRIPTION
**Pull request by hsjoberg**
_Huge thanks to @lucasdcf for helping to write a huge chunk of this BLIP_

Hey everyone, this BLIP is about a simple schema that I've developed for https://github.com/hsjoberg/blixt-wallet, but is better suited as a real standard rather than some proprietary thing.

It provides a structured way to create an invoice description, that is helpful for the user and the wallet software.

### CODE REVIEW

File: blip-0004.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+```
+Paying to: Alfred
+Description: payment for the ice cream you owed me
+```
+
+## Rationale
+
+NameDesc paves a way for a more structued and standardised way of writing
+invoices descriptions. Wallet's that do not support NameDesc will still have a
+graceful fallback, as it will show:
```
Comment (t-bast) [inline blip-0004.md:59]: nit:
```suggestion
NameDesc paves a way for a more structured and standardised way of writing
invoices descriptions. Wallets that do not support NameDesc will still have a
graceful fallback, as they will show:
```

**Review by t-bast — COMMENTED** [review]
LGTM, that's simple enough!

File: README.md [hunk 2]
```diff
@@ -22,3 +22,4 @@ For more detail on the process, please read [bLIP-0001](./blip-0001.md) and
 | [1](./blip-0001.md)      | bLIP Process              | Ryan Gentry                 | Active |
 | [2](./blip-0002.md)      | Reserved Values           | Bastien Teinturier          | Active |
 | [3](./blip-0003.md)      | Keysend                   | Valentine Wallace           | Active |
+| [4](./blip-0004.md)      | NameDesc                  | Hampus Sjöberg              | Active |
```
Comment (ryanthegentry) [inline README.md:25]: rename to blip-0011

File: blip-0004.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+```
+bLIP: 0004
```
Comment (ryanthegentry) [inline blip-0004.md:2]: renumber to 0011

**Review by ryanthegentry — COMMENTED** [review]
Hi @hsjoberg, thanks so much for documenting! Just merged [blip-0010](https://github.com/lightning/blips/pull/10) so my only nits are renumbering and renaming this doc to blip-0011, following Github's numbering convention.

**Review by TheBlueMatt — APPROVED** [review]
ISTM you may want to have some kind of extensibility hooks here - many wallets want to do more than just get a name, and using this may imply you can't add more. Something as simple as suggesting the string `: ` be escaped somehow, eg a double-space, and use `, OtherField: ` to add other fields in the future could suffice. Of course if you feel strongly that you just want to have names, so be it, and this can live as a bLIP and others can extend it in the future.

**Comment by hsjoberg** [comment]
@TheBlueMatt I think you raise a good point regarding extensibility.

NameDesc was intentionally made simplistic to provide an excellent graceful fallback for non-supporting wallets, adding headers kinda ruins this.

For a more advanced use-case, I think a new tag to BOLT11 seems more appropriate.
However if such things get pushback like the [vendor tag](https://github.com/lightning/bolts/pull/694) did (which the idea of NameDesc actually originated from), I suppose one could extend NameDesc. It's not something I wish to strive for though.

File: blip-0004.md (hunk 1 above)
Comment (hsjoberg) [reply blip-0004.md]: Oops! Thank you @t-bast.

File: README.md (hunk 2 above)
Comment (hsjoberg) [reply README.md]: Thank you @ryanthegentry, fixed!

**Comment by hsjoberg** [comment]
I've rebased against latest master (b00212b) and addressed the feedback

File: blip-0004.md (hunk 3 above)
Comment (hsjoberg) [reply blip-0004.md]: Fixed!

**Comment by sbddesign** [comment]
I appreciate this idea from a human usability perspective. For what it's worth, I support it.

- If a wallet implements support for this BLIP, it can reflect this information in the UI in ways that are useful to the user.
- If a wallet does not implement support for this BLIP, the BOLT-11 description stays in a human readable format, like `Payee: this is what you're paying me for`.

**Review by lucasdcf — APPROVED** [review]
LGTM

**Review by Bosch-0 — APPROVED** [review]
LGTM, very simple even a non-dev like me can understand it in its entirety 😆

**Review by ryanthegentry — APPROVED** [review]

**Review by t-bast — APPROVED** [review]
ACK

**Pull request merged**